src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -28,9 +28,18 @@ trait TypeAssigner {
     }
   }
 
-  def avoid(tp: Type, syms: => List[Symbol])(implicit ctx: Context): Type = {
+  /** An upper approximation of the given type `tp` that does not refer to any symbol in `symsToAvoid`.
```

Comment:
This looks fine but is unrelated to the current pull request and ideally should be split in another commit.